### ISSUE TITLE
UX usability issues 

### ISSUE THREAD

**Issue by icnikerazah** [godotengine/godot#16070]
**Godot version:** Godot 3.0 rc3

### Catch Me If You Can
![b](https://user-images.githubusercontent.com/33985414/35413640-2ab4ee38-0231-11e8-8058-58ebfa78e12c.gif)

### Same icon has different function (The first one does not work and I do not know what it does)

<img width="671" alt="ekran resmi 2018-01-26 00 09 29" src="https://user-images.githubusercontent.com/33985414/35413734-755f7c96-0231-11e8-862b-3a475d0ce1f5.png">

### It looks like a key to return to parent directory.

![d](https://user-images.githubusercontent.com/33985414/35413624-1b34541c-0231-11e8-99ee-3cbde70025b8.gif)

### Too many arrows. (Left right arrows are working like a time machine.)

![c](https://user-images.githubusercontent.com/33985414/35414179-d0692f3c-0232-11e8-9002-d7920752c1da.gif)

![ekran resmi 2018-01-26 00 12 02](https://user-images.githubusercontent.com/33985414/35413528-cc909460-0230-11e8-99e0-e828f3da98dd.png)

### It has to be this way;

![ssd](https://user-images.githubusercontent.com/33985414/35414800-a79b03f8-0234-11e8-9903-2a7c84dd58c1.gif)

**Minimal reproduction project:**

**Comment by Calinou** [comment]
I don't like the Apple approach to file management very much; what if we had clickable sections for the path instead of a text file path?

See an example below (in Dolphin):

![screenshot_20180125_232723](https://user-images.githubusercontent.com/180032/35415838-55c2e1b6-0227-11e8-93ca-793b77c4f59e.png)

**Comment by icnikerazah** [comment]
@Calinou 

> I don't like the Apple approach to file management very much (I've always disliked using the Finder); what if we had breadcrumbs (clickable sections) for the path instead of a text file path?

Finder already has these features.
<img width="190" alt="ekran resmi 2018-01-26 01 37 45" src="https://user-images.githubusercontent.com/33985414/35416381-029428b6-023a-11e8-98b0-bce1189515b9.png">
![ssd2](https://user-images.githubusercontent.com/33985414/35416391-097e0872-023a-11e8-95a5-e9c963c456dc.gif)

UX design is not just about pleasure.  Visual Communication is a disipline. simplicity and usability come first.

**Comment by NathanWarden** [comment]
I'm a fan of breadcrumbs from the perspective of a smooth feeling interface. However, they actually hamper usability in some cases. For instance, consider the difference between a text path and breadcrumbs:

Breadcrumbs:
* Can click between sections of a path quickly
* **Can't** copy and paste a path from the file manager to get to that location instantly

Editable text based path:
* Can navigate between sections of the path (albeit not quite as quickly as breadcrumbs)
* **Can** copy and paste a path from the file manager to get to that location instantly

Overall, as far as usability goes, breadcrumbs I find far less usable, especially when I'm trying to get to a secondary projects drive. Also, if you have a lot of folders in the directory you're trying to get to it's often much quicker to type the folder name than to scroll down and look for it.

With that said, I have seen file managers that allow for both, and it could be via an editor option or even better would be via a toggle button (or contextual menu) if it doesn't make it too cluttered. I think that would be the ideal solution.

**Comment by Calinou** [comment]
> Can't copy and paste a path from the file manager to get to that location instantly

I forgot to mention it, but Dolphin allows clicking on empty space to the right of the breadcrumb to turn the breadcrumb into a text path.

**Comment by NathanWarden** [comment]
I think the back and forward buttons are working exactly the same as they do on macOS. If you click on one location and then a completely different location, doesn't the back button go to the previous location, not the parent folder? It's working just like a web browser, which I think is fine.

The up arrow takes you to the parent directory, which also makes sense.

The circular arrow button refreshes the file manager in case there are changes.

All of these functions make sense, but I would 100% agree that maybe the icons could be much more descriptive/less confusing.

**Comment by NathanWarden** [comment]
> I forgot to mention it, but Dolphin allows clicking on empty space to the right of the breadcrumb to turn the breadcrumb into a text path.

@Calinou Haha, I was thinking of mentioning that, but couldn't remember for sure since I haven't used dolphin in about a year. :)

**Comment by djrm** [comment]
the refresh button is pretty standard, about the first issue, yeah it is really annoying, there could be a parent folder icon, to avoid the up arrow (i think the up arrow is used on some programs, but it might be just user preference).

i think the only ux issue here is the first one, maybe the second one, there rest seem to be something that the user can learn and might not be a clear way to do without affecting other user (maybe non mac users have different opinions).

**Comment by vnen** [comment]
> (i think the up arrow is used on some programs, but it might be just user preference).

For reference, the up arrow is used on Windows Explorer.

The first issue here is already tracked on #16027. The refresh icon being the same as undo is a common thing (because they are viewed in different contexts), but maybe it can improved to avoid confusion. The rest is mostly "getting used to it", though I like the idea of breadcrumbs.

**Comment by icnikerazah** [comment]
@vnen 
I did not say what I wanted a new feature. It's all a result of inconsiderate designs. and you are telling us to get used to this bad design.

I did not even talk about the small buttons very close to each other.  this #16027 solve the problem temporarily.

<img width="184" alt="ekran resmi 2018-01-26 03 46 51" src="https://user-images.githubusercontent.com/33985414/35419957-ff7dbd42-024b-11e8-84c6-cd4801c70916.png">

their solution is another discussion topic.

**Comment by vnen** [comment]
> I did not say what I wanted a new feature. It's all a result of inconsiderate designs. and you are telling us to get used to this bad design.

You misunderstood what I said. The icon design is very similar to Windows Explorer, and also relatable to many file managers for Linux. You are too used to the Finder from macOS and thus having trouble getting used to a few different icons in the interface, but that's mostly a personal matter, it's impossible to make the interface appeal every user. We have to make a few compromises.

For instance:

> (Left right arrows are working like a time machine.)

That's consistent within the Godot interface **and** in respect to most file managers (maybe even with macOS as Nathan pointed out). So (at least in this regard) it's not "bad design", it's just a feature you're not used to.

The buttons on the interface are missing tooltips, and that's something that should be fixed. But we have no obligation to make it look and act *exactly* like the Finder.

**Comment by icnikerazah** [comment]
maybe you are right. path navigation part is not very important. but sliders are epic fail.

**Comment by lmjk** [comment]
If you add a point at the end of the arrow, it might be appear less ambiguous.
![godot-reet-preview](https://user-images.githubusercontent.com/25223253/35443579-4e3cc79e-02b4-11e8-85ed-c0dbdf61e515.png)

**Comment by Dillybob1992** [comment]
Just to note: The first s/s, the reset icon appears because you changed an exported var's property. It's useful if you want to set it back to its default value.

**Comment by icnikerazah** [comment]
### When "Return" key is pressed, the input box must be deactivated. this behavior causes an extra click.
![aa](https://user-images.githubusercontent.com/33985414/35445479-cb8675ca-02c2-11e8-9aab-af9b2cd9a5c6.gif)

**Comment by Dillybob1992** [comment]
@icnikerazah That might be intended, because you're canceling the edit box (input has to propagates). In fact, it could be argued it's useful because when a user wants to close the edit box, they just click anywhere outside of it instead of *having to click on something* that doesn't have an action

**Comment by icnikerazah** [comment]
@Dillybob1992 You do not have to say things that everyone is aware of. I do not understand why I'm clicking extra when I have "Return".

**Comment by Dillybob1992** [comment]
@icnikerazah Because when you press return/enter, you are changing that value. How does the editor know, you do / don't want to edit `y` as well?

**Comment by icnikerazah** [comment]
@Dillybob1992 I can click on the box at any time. I can press the tab key. I do not understand what you're talking about.

**Comment by Dillybob1992** [comment]
Well, in your last screenshot, the extra click is needed because it cancels the box. I'm saying that could be argued as an intentional design decision.

 Are you suggesting to close the edit box on enter pressed? That'd be cool

**Comment by icnikerazah** [comment]
Do not you see that I already suggested in my first post.

> When "Return" key is pressed, the input **box must be deactivated**. this behavior causes an extra click.

**Comment by NathanWarden** [comment]
I find the behavior of editing these fields a bit painful too. I wonder if these could be "in-lined", meaning no extra view pops up. You just edit it directly in the inspector without any view popping up. This would also allow tabbing between multiple fields. So, on a Spatial node if you tab from Translation's z it would go to Rotation's x.

But, even allowing "Enter" to just close the view would be a huge plus :) Right now you have to hit "Enter", "Cancel"

**Comment by Dillybob1992** [comment]
Yeah, I just thought that entered didn't close it because the editor didn't know if you're going to edit y either. But yeah, then you mentioned we can **press tab to alternate**, then press enter to finish. I'm just trying to discuss to understand your viewpoints, that's all :)

**Comment by icnikerazah** [comment]
When the "Enter" correction is performed. you can continue to do the following;

![33](https://user-images.githubusercontent.com/33985414/35452456-97e52332-02d7-11e8-8251-4f75a72f783a.gif)

Tab

![ff](https://user-images.githubusercontent.com/33985414/35452457-99285430-02d7-11e8-8e48-68434ee89358.gif)

**Comment by NathanWarden** [comment]
@icnikerazah I like the idea of them being embedded, and I don't really see the point in hitting enter at all to be honest :)

**Comment by Dillybob1992** [comment]
Definitely against embedded values. Takes up way too much space on the Inspector.

**Comment by icnikerazah** [comment]
@Dillybob1992 

> Definitely against embedded values. Takes up way too much space on the Inspector. UI is cramped as is, no need to expand panels further and further I feel :(

Up and down arrows take up a lot of space. they can be abandoned.

![compact](https://user-images.githubusercontent.com/33985414/35476410-68469450-03c0-11e8-89d5-818a8c8353dd.jpg)

**Comment by NathanWarden** [comment]
@icnikerazah That looks much more efficient to work with :) The undos could probably just be recorded whenever a field loses focus.

**Comment by vnen** [comment]
There's quite a bit of contradiction in this thread IMO. First the "Revert" icon looks confusing, but then an arbitrary color change is suggested, which is  more confusion. The color change is meaningless in its own since it's not standard behavior. I have seen bold used for that, but not sure if it's the best fit. Also, double-clicking to revert is also non-standard behavior, would take ages to beginners before they realized this is possible (they'll probably figure out when they revert a value by mistake and get frustrated).

Then there's the problem with small buttons but the solution is to cram everything directly into the Inspector, making the fields smaller and harder to click. And to reduce the font, making it even smaller. Probably a lot of people will complain if the arrow buttons are removed too.

I have to say I do like the idea of showing the components of a vector directly, But we have to consider people with small screens (there's lots of complaints in this regard already). Editing a full transform matrix with a narrow Inspector would be a nightmare.

I would suggest to only show the fields directly if the Inspector is larger than a certain widget, hiding them into the popups if it's too narrow. Still show the revert and keyframe buttons when applicable, but let's find a place where they don't interfere with input boxes. Don't reduce the font size, is already small enough. Keep the arrows in the spin box, they give fine granularity to change values with the mouse.

**Comment by icnikerazah** [comment]
@vnen 
Do you have a constructive idea? You're ignoring the problems I mentioned. Keep it as it is. I'm just throwing ideas seeds. You can close the title if you want.

**Comment by volzhs** [comment]
I was thinking about this but I could find a good solution for it.
and @reduz will rewrite inspector panel at some point and this will help him I guess.

**Comment by Dillybob1992** [comment]
@icnikerazah Might be language barrier, but to me, seems like you are being very passive with your tone. Not sure if that is intended or not :P

**Comment by icnikerazah** [comment]
@Dillybob1992 
Yes My English is not good. I try to explain very simple.
(Neyse bu başlığa yazmaktan sıkıldım mna koym. yöneticiler muhafazakar. siklerinde bile değil)

**Comment by Dillybob1992** [comment]
No problem, nvm me then ^_^ 👍

**Comment by icnikerazah** [comment]
@vnen 
> Also, double-clicking to revert is also non-standard behavior

Double click and Ctrl+Click is a very common and standard usage for Resetting. A few examples;

double-click
![kaydirma](https://user-images.githubusercontent.com/33985414/35477902-996dd876-03df-11e8-801e-2af714e7cf86.gif)

Ctrl+click
![kaydirma2](https://user-images.githubusercontent.com/33985414/35477911-b32970e0-03df-11e8-93b4-c8f0892936dc.gif)

Ctrl-Click
![kay33dirma2](https://user-images.githubusercontent.com/33985414/35477913-b8ae7e02-03df-11e8-8240-ea0ba9fcad16.gif)

You're talking too sure of everything.

**Comment by vnen** [comment]
@icnikerazah Well, I posted a lot of text to express my thoughts. The language barrier kind of gets in the way. You might have missed that I agreed with you in *some* points (read the last paragraph off my last post at least).

> Double click and Ctrl+Click is a very common and standard usage for Resetting. A few examples;

**All** of those are on **sound software** and it's common there for knob controls, especially volume and panning. So I even proposed that for volume control on the audio buses. Show me an example that is not on a sound software. Show me an example of a number field or text field.

----

Yes, I'm against radical changes because:

1. Users already know this interface. If you change too much, everybody will have to re-learn.
2. Likely some parts will be worse for some users. If they are used to other kind of software, they'll bump into an interface hard to understand.
3. All of this thread is based on your personal opinion. Other people may agree or disagree.

**Designing user interface is hard. I agree it can be better, I don't agree we need to remake everything.**

**Comment by icnikerazah** [comment]
@vnen 
mouse over + backspace (blender)
![sss](https://user-images.githubusercontent.com/33985414/35478103-b9a2ffda-03e5-11e8-9a2f-81098136890b.gif)

**Comment by volzhs** [comment]
IMO, it's better to show reset button properly.
I've used blender about a year but never knew backspace resets value. (thanks for the blender tip)

**Comment by icnikerazah** [comment]
@volzhs 

> IMO, it's better to show reset button properly.
I've used blender about a year but never knew backspace resets value. (thanks for the blender tip)

Is the Godot Designed for children with learning disabilities? Can you be able to touch the touchscreen phone?

**Comment by volzhs** [comment]
I agree with inspector panel should be upgraded to give better usability.
But I think, something should be shown in a clear way, like resetting value with button.

can you find a functionality if it's not shown on screen and described about it?
as I told you about blender resetting value, I never knew there's such a functionality.
I think it's bad design.

**Comment by volzhs** [comment]
Good design is to make it possible to use even if YOU ARE DISABLED, not teaching in your way.
![image](https://user-images.githubusercontent.com/8281454/35482649-afdff926-047b-11e8-99a0-4280b547b945.png)

**Comment by icnikerazah** [comment]
It is better to compare Godot with the useless linux interfaces. Not Apple.

**Comment by hpvb** [comment]
@icnikerazah We appreciate your input but please keep your comments constructive. We're all trying to make the best possible game engine here.

**Comment by icnikerazah** [comment]
you delete my post. you fucking hitler.

**Comment by icnikerazah** [comment]
what the fuck. hpvb is deleting my posts

**Comment by Noshyaar** [comment]
Last warning. If you keep using bad language, this issue will be locked.

**Comment by reduz** [comment]
Calm down guys, this thread has many good ideas. As I mentioned before, I will be rewriting the inspector, lkely after 3.1 is out (priorities..) and I would like to use this thread as reference. 

We will also likely have the chance to try out many ideas by then.

**Comment by reduz** [comment]
Just be patient until then.

**Comment by icnikerazah** [comment]
you dont have right to delete my post. you fucking geeks

**Comment by reduz** [comment]
@icnikerazah calm down, no one should use abusive language, it's not something personal with you.
